test/controllers/declaration/DeclarationHolderChangeControllerSpec.scala

```diff
@@ -133,25 +133,56 @@ class DeclarationHolderChangeControllerSpec extends ControllerSpec with OptionVa
         "user edit leads to duplicate data" in {
           withNewCaching(aDeclarationAfter(request.cacheModel, withDeclarationHolders(declarationHolder1, declarationHolder2)))
 
-          val requestBody =
-            Seq(
-              "authorisationTypeCode" -> declarationHolder2.authorisationTypeCode.get,
-              "eori" -> declarationHolder2.eori.map(_.value).get,
-              "eoriSource" -> declarationHolder2.eoriSource.map(_.toString).get
-            )
+          val requestBody = List(
+            "authorisationTypeCode" -> declarationHolder2.authorisationTypeCode.get,
+            "eori" -> declarationHolder2.eori.map(_.value).get,
+            "eoriSource" -> declarationHolder2.eoriSource.map(_.toString).get
+          )
           val result = controller.submitForm(Mode.Normal, declarationHolder1.id)(postRequestAsFormUrlEncoded(requestBody: _*))
 
           status(result) mustBe BAD_REQUEST
           verifyChangePageInvoked()
         }
+
+        "user adds mutually exclusive data" when {
```

Comment:
IMHO these tests should be done for the `DeclarationHolderHelper` class or `Form` class if you decide to move the validation there.
It is not controller's  responsibility to validate the input, only to invoke that validation. Besides, right now the test cases are duplicated.